beater/server.go

```diff
@@ -62,18 +62,26 @@ func newServer(config Config, publish successCallback) *http.Server {
 	}
 }
 
-func start(server *http.Server, ssl *SSLConfig) {
-	if ssl.isEnabled() {
-		go server.ListenAndServeTLS(ssl.Cert, ssl.PrivateKey)
-	} else {
-		go server.ListenAndServe()
-	}
+func start(server *http.Server, ssl *SSLConfig) chan (error) {
+	c := make(chan (error), 1)
+
+	go func() {
+		if ssl.isEnabled() {
+			c <- server.ListenAndServeTLS(ssl.Cert, ssl.PrivateKey)
+		} else {
+			c <- server.ListenAndServe()
+		}
+	}()
+
+	return c
 }
 
-func stop(server *http.Server) error {
+func stop(server *http.Server) {
 	c := context.TODO()
 	err := server.Shutdown(c)
-	return err
+	if err != nil {
```

Comment:
I would leave it to the caller to decide if an error should be logged or not. This also makes it easier for testing to see if an error was returned or not.